### PR TITLE
Fixed bug with incorrect UI state after activity recreate

### PR DESCRIPTION
**Pull request by silentnuke**
## Overview :notebook:  

After screen rotation or process death `MultipleBackStacks` state is not restored properly. If `NavController.OnDestinationChangedListener` is set, after activity recreation it's not set and navcontroller doesn't have any active listener. This leads to different problems, for e.g. if `NavController.OnDestinationChangedListener` is used to hide some views depending on destitation it doesn't work as expected. Also `onBottomNavigationView` returns `LiveData` which instead of controller contains null.

Intorduced a new screen which hides bottom menu and allows to write an instrumeted test to verify fix.
Also did a minor change to bottom menu to highlight currently selected item ;-)

## Screenshots, video :framed_picture:

| Before | After |
|--------|-------|
|ezgif-7-0434d1f7dc03|ezgif-7-9338fc78e6d2|

## Testing :mag:
Added new UI test to cover that case, also verified that all previous tests are passing

### CODE REVIEW

**Comment by silentnuke** [comment]
@VictorAlbertos, @mbarta hey guys, found another issue and prepared fix for it, PTAL.
PS: not sure why, but it's not possible to request review or assign, is it expected?
image

**Comment by VictorAlbertos** [comment]
Thanks again for this new contribution! I'm going to review the PR, but it looks pretty useful at first glance :)

> PS: not sure why, but it's not possible to request review or assign, is it expected?

No idea, we have the default setup, but in any case, it is fine as it is because we get the notification and we internally assign some member to the PR ;)

File: app/src/main/java/cookpad/com/bottomnavwatson/DetailWithoutBottomMenuFragment.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavDeepLinkBuilder
+import androidx.navigation.fragment.NavHostFragment.findNavController
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupWithNavController
+import kotlinx.android.synthetic.main.detail_fragment.toolbar
+import kotlinx.android.synthetic.main.detail_fragment.tvTestDeepLinks
+import java.util.Random
+
+class DetailWithoutBottomMenuFragment : Fragment(R.layout.detail_without_bottom_menu_fragment) {
```
Comment (VictorAlbertos) [inline app/src/main/java/cookpad/com/bottomnavwatson/DetailWithoutBottomMenuFragment.kt]: As `DetailWithoutBottomMenuFragment` is almost identical to `DetailFragment`, please use the `SafeArgs` to supply a parameter called `hideNavBar` or alike based on which the layout and the visibility of the navbar will be set.

File: bottom-nav-watson/src/main/kotlin/com/cookpad/watson/MultipleBackStacks.kt [hunk 2]
```diff
@@ -61,6 +64,14 @@ internal class MultipleBackStacks(
         return selectedNavController
     }
 
+    private fun restoreNavController(destinationChangedListener: NavController.OnDestinationChangedListener?) {
```
Comment (VictorAlbertos) [inline bottom-nav-watson/src/main/kotlin/com/cookpad/watson/MultipleBackStacks.kt:68]: Nice 👍

**Review by VictorAlbertos — APPROVED** [review]
LGTM! Please address my suggestion about removing `DetailWithoutBottomMenuFragment` and reusing `DetailFragment` instead.

**Review by mbarta — APPROVED** [review]
Thanks for another bug fix! :+1:

File: app/src/main/java/cookpad/com/bottomnavwatson/DetailWithoutBottomMenuFragment.kt (hunk 1 above)
Comment (silentnuke) [reply app/src/main/java/cookpad/com/bottomnavwatson/DetailWithoutBottomMenuFragment.kt]: Done

**Comment by silentnuke** [comment]
@VictorAlbertos addressed your feedback, also found another issue, `selectedFragmentTag` was not initialized during first start, fixed it too.

**Comment by VictorAlbertos** [comment]
Thanks for addressing the suggestion 👍

**Pull request merged**